src/classes/CRLP_RollupProcessor_TEST.cls

```diff
@@ -0,0 +1,314 @@
+                AccountId = account.Id,
+                Amount = 100,
+                CloseDate = Date.today(),
+                StageName = UTIL_UnitTestData_TEST.getClosedWonStage()
+        );
+
+        OpportunityContactRole ocr1 = new OpportunityContactRole(
+                Id = UTIL_UnitTestData_TEST.mockId(OpportunityContactRole.SObjectType),
+                OpportunityId = opp.Id,
+                ContactId = contact.Id,
+                Role = UTIL_CustomSettingsFacade.DEFAULT_OPPORTUNITY_CONTACT_ROLE_DONOR,
+                IsPrimary = false
+        );
+
+        Partial_Soft_Credit__c psc = new Partial_Soft_Credit__c(
+                Id = UTIL_UnitTestData_TEST.mockId(Partial_Soft_Credit__c.SObjectType),
+                Opportunity__c = opp.Id,
+                Opportunity__r = opp,
+                Contact__c = contact.Id,
+                Contact__r = contact,
+                Role_Name__c = UTIL_CustomSettingsFacade.DEFAULT_OPPORTUNITY_CONTACT_ROLE_DONOR,
+                Contact_Role_Id__c = ocr1.Id,
+                Amount__c = opp.Amount
+        );
+
+        CRLP_RollupProcessingOptions.ProcessingOptions options = new CRLP_RollupProcessingOptions.ProcessingOptions();
+        options.useRollupDefsByParentIdMap = true;
+
+        CRLP_RollupProcessor processor = new CRLP_RollupProcessor()
+                .withBatchJobMode(CRLP_RollupProcessingOptions.BatchJobMode.SkewMode)
+                .withRollupType(CRLP_RollupProcessingOptions.RollupType.ContactSoftCredit)
+                .withSummaryRecord(contact)
+                .withDetailRecords(new List<SObject> { psc })
+                .withJobOptions(options);
+
+        System.assertEquals(OpportunityContactRole.SObjectType, processor.detailObjectType);
+        System.assertEquals(UTIL_Namespace.StrAllNSPrefix('Contact__c'), processor.parentKeyField);
+        System.assertEquals(UTIL_Namespace.StrAllNSPrefix('Contact__c'), processor.parentRelationshipField);
+        System.assertEquals(null, processor.parentRelationshipObject);
+
+        System.assertEquals(1, processor.detailRecordsByParentId.get(contact.Id).size());
+
+        System.assertEquals(true, processor.isSkewMode);
+        System.assertEquals(true, processor.isOppContactRoleSoftCreditRollup);
+
+        // Run the Account through the Skew Mode processor.
+        List<SObject> updatedRecords = processor.startRollupProcessing();
+
+        System.assert(updatedRecords.isEmpty());
+        System.assertEquals(1, processor.getRollupDefsForParent(contact.Id).size());
+        System.assertNotEquals(null, processor.getLastParentIdProcessed());
+        System.assertEquals(true, processor.hasLastParentIdCompleted());
+
+        List<CRLP_Rollup> previousRollupDefs = processor.getRollupDefsForParent(contact.Id);
+
+        // In Skew Mode, the last Contact processed has to be sent back through the
+        // completeRollupForSingleSummaryRecord() method.
+        processor = new CRLP_RollupProcessor()
```

Comment:
Is this test verifying more than one concept (we needed to rebuild the `processor`)?

Reply:
No. Skew Mode is a two-part operation. There's the rollup of data and then there is the completion of the rollup operation (i.e. the call to `completeRollupForSingleSummaryRecord`). The first section calls `startRollupProcessing()` to perform the rollup operation. Because it's skew mode and we're only processing one record, the first part only rolls up the data and returns a modified `List<CRLP_Rollup>` collection. The second call to `completeRollupForSingleSummaryRecord` is needed to complete the rollup and actually return an updated SObject instance.